src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -9993,6 +9993,35 @@ public static void M()
             VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(src, expectedFlowGraph, expectedDiagnostics, parseOptions: TestOptions.RegularPreview);
         }
 
+        [Fact, WorkItem(53849, "https://github.com/dotnet/roslyn/issues/53849")]
+        public void WithExpr_AnonymousType_ValueIsLoweredToo()
+        {
+            var src = @"
+var x = new { Property = 42 };
+var adjusted = x with { Property = x.Property + 2 };
+
+System.Console.WriteLine(adjusted);
+";
+            var comp = CreateCompilation(src, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyEmitDiagnostics();
```

Comment:
>VerifyEmitDiagnostics [](http://example.com/codeflow?start=17&length=21)

Minor: Calling VerifyDiagnostics on result of CompileandVerify might be more efficient (avoids doing the work twice).